### PR TITLE
Link documentation from console

### LINKED ISSUE #2254: Referring to documentation from within the console
#### Summary
We need to be able to refer to our documentation from within the console.

#### Why do we need this?
Provide help right where it is needed.

#### What is already there? What do you see now?
Documentation

#### What is missing? What do you want to see?
A safe way to refer to the documentation (linking) from within the console.

#### How do you propose to implement this?
- Extend the console configuration to provide a base path to the documentation
- Implement a `<DocLink />` component that will concatenate the base path with the documentation path

#### Can you do this yourself and submit a Pull Request?
@pgalic96

### PR DESCRIPTION
**Pull request by pgalic96**
#### Summary

Closes #2254 

#### Changes

- Pass documentation url to the console
- Introduce `DocLink` component which appends `path` to the documentation url
- Introduce `generateDocUrl` util for generating the documentation url

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: cmd/internal/shared/console/config.go [hunk 1]
```diff
@@ -43,6 +47,7 @@ var DefaultConsoleConfig = console.Config{
 			JSFiles:       []string{"console.js"},
 		},
 		FrontendConfig: console.FrontendConfig{
+			DocumentationBaseURL: fmt.Sprintf("https://thethingsstack.io/%s", strings.TrimSuffix(strings.TrimPrefix(version.TTN, "v"), "-dev")),
```
Comment (johanstokking) [inline cmd/internal/shared/console/config.go:50]: I thought we were getting rid of publishing versioned docs? cc @htdvisser

Comment (htdvisser) [reply cmd/internal/shared/console/config.go:50]: 1. Documentation links in the console should point to the exact version of the console that the user is currently working with.
2. At the moment this PR is submitted we still have versioned docs.

File: pkg/webui/components/link/index.js [hunk 2]
```diff
@@ -118,6 +119,69 @@ Link.defaultProps = {
...
+  intl: PropTypes.shape({
+    formatMessage: PropTypes.func,
+  }).isRequired,
+  name: PropTypes.string,
+  path: PropTypes.string.isRequired,
+  primary: PropTypes.bool,
+  secondary: PropTypes.bool,
+  showVisited: PropTypes.bool,
+  title: PropTypes.string,
+  titleValues: PropTypes.shape({}),
```
Comment (kschiffer) [inline pkg/webui/components/link/index.js:170]: Better to spread the prop types from the other link component. There's too much duplication.

```js
DocLink.propTypes = {
   ...Link.propTypes,
   …
}
```

File: pkg/webui/components/link/index.js [hunk 3]
```diff
@@ -118,6 +119,69 @@ Link.defaultProps = {
...
+  className: undefined,
+  id: undefined,
+  disabled: false,
+  title: undefined,
+  name: undefined,
+  primary: false,
+  titleValues: undefined,
+  secondary: false,
+  showVisited: false,
+}
```
Comment (kschiffer) [inline pkg/webui/components/link/index.js:183]: Same here.

**Comment by pgalic96** [comment]
@kschiffer I changed the `DocLink` component to use whatever config is present in `EnvProvider`. However, for the `generateDocUrl` util I think this is the simplest solution to retrieve the `documentation_base_url`, just fetching the config in `pkg/webui/lib/env` will crash the storybook since it does not have `env` variables in template data. Suggestions on how can this util function use `env` in context are welcome, but I think this is a satisfying solution.

File: config/storybook/env.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
```
Comment (kschiffer) [inline config/storybook/env.js:1]:
```suggestion
// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
```

**Comment by kschiffer** [comment]
> Suggestions on how can this util function use env in context are welcome, but I think this is a satisfying solution.

Yeah that's good enough.

**Pull request merged**